src/notebook.c

```diff
@@ -754,6 +779,314 @@ void notebook_remove_page(gint page_num)
 }
 
 
+/* Copied from sidebar.c.  We can place this in one place like utils.c. */
+static gboolean utils_filename_has_prefix(const gchar *str, const gchar *prefix)
+{
+	gchar *head = g_strndup(str, strlen(prefix));
+	gboolean ret = utils_filenamecmp(head, prefix) == 0;
+
+	g_free(head);
+	return ret;
+}
+
+
+/* Copied from sidebar.c.  We can place this in one place like utils.c,
+ * with a more formal naming convention. */
+static gchar *get_doc_folder(const gchar *path)
```

Comment:
It will already be totally different if you use "sort by basename"; and I find it kinda odd to want to see the sidebar *and* have the tabs sorted (which indeed would be consistent, but would be totally redundant).

I don't have a strong opinion here, I just found it a kinda artificial requirement that made the code uglier than needed.